### ISSUE TITLE
Unable to use with 2FA

### ISSUE THREAD

**Issue by jimmyeao** [OfficeDev/O365-InvestigationTooling#8]
Script is not compatible with 2 factor authentication

**Comment by paulschnack** [comment]
No it doesn't, would be great if it could be modified so it did.

**Comment by alexlushsevernunival** [comment]
I find it amusing that MS link to this script from Office365 Secure Score. One of the top rules that Secure Score suggests first is to 'enable Multi-Factor auth for ALL admins'. Assuming you have followed their advise on that rule, you are then unable to run this script to check mailbox delegation as it doesn't support MFA!!!

**Comment by citykidz** [comment]
Assuming you're using the latest Microsoft Exchange Online Powershell module (and you'll know you are, because it will allow you to login using MFA), then just copy and run everything from line 9 and forward.

**Comment by knibbscs** [comment]
Works fine with 2FA - you just need to create and use an app password to sign in.

**Comment by sbdoroff** [comment]
@knibbscs - not able to get to work with app pass

**Comment by domcote** [comment]
yeah - go here to get the PS Module that uses MFA: https://technet.microsoft.com/en-us/library/mt775114(v=exchg.160).aspx. Then follow the instructions to connect at the top of the shell - PS:Connect-EXOPSSession…. works fine!
